consumer/consumermock/mock_consumers.go

```diff
@@ -0,0 +1,183 @@
+package consumermock
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+)
+
+var _ consumer.TraceConsumer = (*Trace)(nil)
+var _ consumer.TraceConsumerOld = (*Trace)(nil)
+
+type Trace struct {
+	sync.Mutex
+	spansReceived     atomic.Uint64
+	receivedTraces    []pdata.Traces
+	receivedTracesOld []consumerdata.TraceData
+}
+
+func (tc *Trace) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+	tc.spansReceived.Add(uint64(td.SpanCount()))
+
+	rs := td.ResourceSpans()
+	for i := 0; i < rs.Len(); i++ {
+		ils := rs.At(i).InstrumentationLibrarySpans()
+		for j := 0; j < ils.Len(); j++ {
+			spans := ils.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				var spanSeqnum int64
+				var traceSeqnum int64
+
+				seqnumAttr, ok := span.Attributes().Get("load_generator.span_seq_num")
+				if ok {
+					spanSeqnum = seqnumAttr.IntVal()
+				}
+
+				seqnumAttr, ok = span.Attributes().Get("load_generator.trace_seq_num")
+				if ok {
+					traceSeqnum = seqnumAttr.IntVal()
+				}
+
+				// Ignore the seqnums for now. We will use them later.
+				_ = spanSeqnum
+				_ = traceSeqnum
+			}
+		}
+	}
+
+	tc.Lock()
+	defer tc.Unlock()
+	tc.receivedTraces = append(tc.receivedTraces, td)
+
+	return nil
+}
+
+// ConsumeTraceData consumes trace data in old representation
+func (tc *Trace) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+	tc.spansReceived.Add(uint64(len(td.Spans)))
+
+	for _, span := range td.Spans {
+		var spanSeqnum int64
+		var traceSeqnum int64
+
+		if span.Attributes != nil {
+			seqnumAttr, ok := span.Attributes.AttributeMap["load_generator.span_seq_num"]
```

Comment:
Not sure what to do with this. Looks specific to testbed but also appears unused.

Reply:
If not used, would recommend starting with removing them from testbed initially, then understand what is left in the testbed.